test.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+from question_model import Question
+from data import question_data
+from quiz_brain import QuizBrain
+
+NR_QUESTIONS_EXPECTED = 12
+SCORE_EXPECTED = 0
+QUESTION_NR_EXPECTED = 0
+
+
+class TestQuizBrain(unittest.TestCase):
+    def setUp(self):
+        question_bank = [Question(q_text=q["text"], q_answer=q["answer"]) for q in question_data]
+        self.quiz = QuizBrain(question_bank)
+
+    def test_initialization(self):
+        self.assertEqual(self.quiz.question_number, NR_QUESTIONS_EXPECTED)
+        self.assertEqual(self.quiz.score, SCORE_EXPECTED)
+        self.assertEqual(len(self.quiz.questions_list), NR_QUESTIONS_EXPECTED)
+
+    def test_still_has_question(self):
+        self.assertTrue(self.quiz.still_has_question())
+        # Assuming the quiz has been set up with 12 questions, let's answer all of them
+        for _ in range(12):
+            self.quiz.next_question()
+        self.assertFalse(self.quiz.still_has_question())  # After answering all questions, there should be none left
+
+    def test_check_answer_correct(self):
```

Comment:
What happens if you have an empty quiz and you try to check an answer? I believe you're going to divide by zero.

Reply:
[394ebdf](https://github.com/rawadabu/quiz-game/pull/1/commits/394ebdf3217e45adbea2bcf9a0f60a0ba620a4b9) You are correct that there is a potential issue if the quiz is empty. However, the code has been updated and I think I've handled this scenario @yftacherzog

Reply:
I would have opted for eliminating the reason for the corner case to exist instead of adding a validation (another `if`, another potential bug, another required test).

In this case, `check_answer()` is suppose to check something that is related to a question on the quiz, but it's not actually doing that. It instead checking arbitrary equality between strings, and this is why you can run it when there are no questions in the quiz.

If you instead have it as a method of `Question`, you won't have to verify that the question actually exists, as it would be impossible to call it without a question.

Reply:
Got it, updating